app/helpers/application_helper.rb

```diff
@@ -8,4 +8,12 @@ def page_title(page)
       t('page_titles.application')
     end
   end
+
+  def page_heading(action, resource)
+    actions = { new: 'Add', edit: 'Edit' }
+
+    raise 'Action not supported' unless actions.include?(action)
```

Comment:
I think it's more idiomatic to use some standard ruby error here with a message, like `ArgumentError`?

Reply:
good point, I found out that using fetch (suggested by @duncanjbrown) generate by default `KeyError`. I think that's perfect, `ArgumentError` probably applies better when calling a method.

Reply:
You need to give it a block — see [the docs](https://apidock.com/ruby/Hash/fetch)

Reply:
hey Duncan, at first I thought so too, but actually, you do not need a block, from the docs: 

```
h = { "a" => 100, "b" => 200 }
h.fetch("z")
```

produces:

```
prog.rb:2:in `fetch': key not found (KeyError)
```

Reply:
Yes, `KeyError` is the default, but you can use the block to throw any error you like

```
{a: 1, b: 2}.fetch(:c) { |missing_key| raise ArgumentError("No action found: #{missing_key}") }
```

Reply:
This was clarified with @duncanjbrown. No block is fine.

Reply:
I prefer `ArgumentError` with a message over `KeyError` in this case, as it's more helpful to the reader. 

I'd also recommend a helpful message as in @duncanjbrown 's example.